util/xray.py

```diff
@@ -69,8 +66,14 @@ def _before_first_request(self):
         request._palace_first_request = True
 
     def _before_request(self):
+        if getattr(self.app, "_xray_first_request_done", None) is None:
+            # First request so we do first request processing
+            setattr(self.app, "_xray_first_request_done", True)
+            self._before_first_request()
+
```

Comment:
I think we can just drop all the `before_first_request` stuff from xray. We should be able to just remove `_before_first_request` completely. All its doing is adding an extra annotation to the first request, so we could filter that request out of the data since it takes significantly longer due to the initialization. Since we're not doing that anymore, no need for the extra annotation.